pkg/commands/git_commands/branch.go

```diff
@@ -109,6 +109,15 @@ func (self *BranchCommands) SetUpstream(remoteName string, remoteBranchName stri
 	return self.cmd.New(fmt.Sprintf("git branch --set-upstream-to=%s/%s %s", self.cmd.Quote(remoteName), self.cmd.Quote(remoteBranchName), self.cmd.Quote(branchName))).Run()
 }
 
+func (self *BranchCommands) UnsetUpstream(branchName string) error {
+	// TODO: how should we handle the error if the branch already has no upstream?
```

Comment:
We should hide the option to unset the upstream if there is no current upstream, but if we happen to get this error anyway (due to some race condition) then I would just surface the error to the user as if it were any other error.

Reply:
We should maybe hide the menu altogether, then? Since there is only one option, but that seems really flimsy from a user'"s perspective - "I swear there was a menu here under 'u'..."

Reply:
I wouldn't hide the menu altogether because you'll still always be able to set a new upstream regardless of whether you can unset an existing one.

And yeah I think that error is fine: it's going to be a rare case where a branch obtains an upstream somehow after we render the menu option.

Reply:
> I wouldn't hide the menu altogether 

Neither would I, but having a menu of just 1 option seems odd when encountered in the wild.

Imagine a new lazygit (and a novice `git`) user, they create a new branch, and see the `u` mapping that opens up a menu to `Set/Unset upstream`, but when they open it, there's only the option to `set upstream`. Sure, it's not that difficult to conclude that you can't `unset`, but my initial reaction would be "Huh, that's weird". Wouldn't it be more intuitive to have both actions there and to let the user know that they can try to `unset`, but it causes an error.

Reply:
Very well, you've convinced me!